memgpt/local_llm/llm_chat_completion_wrappers/configurable_wrapper.py

```diff
@@ -0,0 +1,301 @@
+import json
+from typing import List
+
+import yaml
+
+from .wrapper_base import LLMChatCompletionWrapper
+from ...constants import JSON_ENSURE_ASCII
+from ..json_parser import clean_json
+from ...constants import JSON_ENSURE_ASCII
+from ...errors import LLMJSONParsingError
+
+
+# A configurable model agnostic wrapper.
+class ConfigurableWrapper(LLMChatCompletionWrapper):
+    def __init__(
+        self,
+        pre_prompt: str = "",
+        post_prompt: str = "",
+        sys_prompt_start: str = "",
+        sys_prompt_end: str = "",
+        user_prompt_start: str = "",
+        user_prompt_end: str = "",
+        assistant_prompt_start: str = "",
+        assistant_prompt_end: str = "",
+        tool_prompt_start: str = "",
+        tool_prompt_end: str = "",
+        include_sys_prompt_in_first_user_message: bool = False,
+        default_stop_sequences=None,
+        simplify_json_content: bool = False,
+        strip_prompt: bool = False,
+        json_indent: int = 2,
+        clean_function_args: bool = False,
+    ):
+        """
+        Initializes a new MessagesFormatter object.
+
+        Args:
+            pre_prompt (str): The pre-prompt content.
+            post_prompt(str): The post-prompt content
+            sys_prompt_start (str): The system prompt start.
+            sys_prompt_end (str): The system prompt end.
+            user_prompt_start (str): The user prompt start.
+            user_prompt_end (str): The user prompt end.
+            assistant_prompt_start (str): The assistant prompt start.
+            assistant_prompt_end (str): The assistant prompt end.
+            include_sys_prompt_in_first_user_message (bool): Indicates whether to include the system prompt
```

Comment:
I think one piece of functionality this wrapper is missing that the ChatML wrappers are missing is the ability to have a special prefix for the first message, eg https://github.com/cpacker/MemGPT/blob/main/memgpt/local_llm/llm_chat_completion_wrappers/chatml.py#L42

Reply:
Is added